tianshou/utils/net/common.py

```diff
@@ -9,46 +9,134 @@
 def miniblock(
     inp: int,
     oup: int,
-    norm_layer: Optional[Callable[[int], nn.modules.Module]],
+    norm_layer: Optional[Callable[[int], nn.modules.Module]] = None,
+    activation: Optional[nn.modules.Module] = nn.ReLU,
 ) -> List[nn.modules.Module]:
     """Construct a miniblock with given input/output-size and norm layer."""
     ret: List[nn.modules.Module] = [nn.Linear(inp, oup)]
     if norm_layer is not None:
         ret += [norm_layer(oup)]
-    ret += [nn.ReLU(inplace=True)]
+    ret += [activation(inplace=True)]
     return ret
 
 
-class Net(nn.Module):
+class MLP(nn.Module):
     """Simple MLP backbone.
+    Create a MLP of size inp_shape*hidden_layer_size[0]*hidden_layer_size[1]...
+
+    :param hidden_layer_size: shape of MLP passed in as a list, not incluing
+        inp_shape.
+    :param norm_layer: use which normalization before activation, e.g.,
+        ``nn.LayerNorm`` and ``nn.BatchNorm1d``, defaults to no normalization.
+        You can also pass in a list of hidden_layer_number-1 normalization
+        modules to use different normalization module in different layers.
+        Default to empty list (no normalization).
+    :param activation: which activation to use after each layer, can be both
+        the same actvition for all layers if passed in nn.Module, or different
+        activation for different Modules if passed in a list.
+        Default to nn.ReLU.
+    """
+
+    def __init__(
+        self,
+        hidden_layer_size: Union[int, List[int]] = [],
+        norm_layer: Optional[Union[nn.modules.Module,
+                                   List[nn.modules.Module]]] = [],
+        activation: Optional[Union[nn.modules.Module,
+                                   List[nn.modules.Module]]] = nn.ReLU,
+        inp_shape: Optional[Union[tuple, int]] = None,
+        device: Union[str, int, torch.device] = "cpu",
+    ) -> None:
+        super().__init__()
+        if isinstance(hidden_layer_size, int):
```

Comment:
I would rather not allow such usecase, it is ill-defined and misleading.

Reply:
No, hidden_layer_size as int is not allowed now, I only check this because originally users pass in layer_num(int) as first argument of Net. This is a reminder to them that original class has been changed when they see the error. Again, for consistency.

Reply:
resolved as replacing hidden_layer_size with hidden_sizes